homeassistant/components/netatmo/climate.py

```diff
@@ -441,12 +474,48 @@ async def _async_service_set_preset_mode_with_end_datetime(
             mode=PRESET_MAP_NETATMO[preset_mode], end_time=end_timestamp
         )
         _LOGGER.debug(
-            "Setting %s preset to %s with optional end datetime to %s",
+            "Setting %s preset to %s with end datetime %s",
             self._room.home.entity_id,
             preset_mode,
             end_timestamp,
         )
 
+    async def _async_service_set_temperature_with_end_datetime(
+        self, **kwargs: Any
+    ) -> None:
+        target_temperature = kwargs[ATTR_TARGET_TEMPERATURE]
+        end_datetime = kwargs[ATTR_END_DATETIME]
+        end_timestamp = int(dt_util.as_timestamp(end_datetime))
+
+        _LOGGER.debug(
+            "Setting %s to target temperature %s with end datetime %s",
+            self._room.entity_id,
+            target_temperature,
+            end_timestamp,
+        )
+        await self._room.async_therm_manual(target_temperature, end_timestamp)
+
+    async def _async_service_set_temperature_with_time_period(
```

Comment:
> I also realised my use case (and I expect many other automation use cases) would require the use of jinja to determine the end date & time. It seemed to make sense to have a service to handle time period as well so I've added that.

I think we should probably try to mimic what's in the "official app/device". Do you select an end time or a run time?
We could of course do both services but I wonder if it really makes sense.

Reply:
For setting the value, the app shows the time period as a slider and defaults to 3h. For displaying the current manual boost, it shows the time it is active until. The API works by timestamp.

I added the extra processing required for time period to avoid having to use as much jinja in automations, but I can see a use case for either. e.g. Set temperature for 3h while clothes are drying or set temperature until 5pm when finishing work.

To set until a given time today, jinja would still be required to work out the date, but it would be much simpler than calculating the time period (e.g. `{{ states('sensor.date') + ' 17:00:00' }}`).

Likewise, having a service to boost for a given timespan would potentially satisfy most use cases in the UI. Working only by datetime would involve some complex jinja to calculate the correct end time.

Maybe HA could handle this automatically in future by having the datetime selector work with times relative to the point when they are triggered, but that is a very different feature.

Reply:
I guess with either one implemented, the alternative can become a blueprint script.

How would you like to proceed?

Reply:
For preset we implemented it with and end datetime. What's there in the app?

Reply:
For preset you activate it indefinitely, then can "edit" to a specific date and time. There is no time period option for preset in the app.

Display of preset:
![Screenshot_2023-11-20-11-10-56-24_2ac22f0e18cbfbeda85a7e653b55b2c6 (Small)](https://github.com/home-assistant/core/assets/50599779/b12a0fbe-2106-44e8-8c2e-9cb90b8b0965)

Edit preset:
![Screenshot_2023-11-20-11-11-03-08_2ac22f0e18cbfbeda85a7e653b55b2c6 (Small)](https://github.com/home-assistant/core/assets/50599779/7c893350-39c0-4aff-987d-02740d841871)

Edit temperature (select time period but end time also visible):
![Screenshot_2023-11-20-11-11-14-73_2ac22f0e18cbfbeda85a7e653b55b2c6 (Small)](https://github.com/home-assistant/core/assets/50599779/7467335d-4e6f-427f-9d7b-3694c851277b)

Display manual temperature:
![Screenshot_2023-11-20-11-11-19-98_2ac22f0e18cbfbeda85a7e653b55b2c6 (Small)](https://github.com/home-assistant/core/assets/50599779/2f0940fb-e85b-4436-a564-4033c6dfa3d9)

Reply:
I guess it can make sense to have both then. Thanks